### ISSUE TITLE
issue running in docker on Sonoff iHost

### ISSUE THREAD

**Issue by roger-richards** [Luligu/matterbridge#175]
Allow me to elaborate on my environment and use case: I use a [Sonoff iHost](https://sonoff.tech/product/gateway-and-sensors/ihost/) to integrate devices into Apple Home.  iHost, in addition to native functionality which integrates a variety of Zigbee devices, it includes a docker hosting feature. iHost also includes a Matter Bridge for the devices it supports. For instance, I am using iHost to integrate Zigbee IKEA roller shade (without using an IKEA hub) and Zigbee Sonoff light controller and expose it to Apple Home via the integrated Matter Bridge.  I am also running [Homebridge](https://homebridge.io/) in a iHost docker to integrate various WiFi devices to Apple Home. Shelly devices make up a large part of my home automation.  I've integrated them via two Homebridge plugins but this does not seem to be a supported approach going forward (Shelly Gen3 devices are not supported by any Homebridge plugin.)  Shelly recommends matterbridge and has a robust matterbridge-shelly plugin.  In order to simplify my Shelly integration and add support for Gen3 devices, I installed matterbrige in an iHost docker. Added matterbridge-shelly via the web gui.  It discovers all my Shelly devices.  Before I add this matterbridge to my Apple Home, I noticed the two devices integrated via native iHost matter bridge (IKEA shade & Sonoff switch) stopped responding in Apple Home.  I stopped the matterbridge docker and these two devices responded again.  There seems to be a conflict between iHost Matter Bridge and matterbridge running in docker on iHost. The only log error in the matterbridge log is:

```
info - [10:14:16.700] [CaseServer]: Received pairing request from udp://fe80::1ccb:403f:2375:5f7%eth0:49956
error - [10:14:16.705] [CaseServer]: An error occurred during the commissioning Fabric cannot be found from destinationId at FabricManager.findFabricFromDestinationId (file:///app/node_modules/@matter/protocol/dist/esm/fabric/FabricManager.js:184:11) at CaseServer.handleSigma1 (file:///app/node_modules/@matter/protocol/dist/esm/session/case/CaseServer.js:115:36) at async CaseServer.onNewExchange (file:///app/node_modules/@matter/protocol/dist/esm/session/case/CaseServer.js:36:7) at async SecureChannelProtocol.onNewExchange (file:///app/node_modules/@matter/protocol/dist/esm/securechannel/SecureChannelProtocol.js:122:9) at async ExchangeManager.onMessage (file:///app/node_modules/@matter/protocol/dist/esm/protocol/ExchangeManager.js:207:9)
info - [10:14:16.710] [InsecureSession]: End insecure session insecure/9412610887785303301
```

**Comment by Luligu** [comment]
Hi,
first of all thank you for your interest in Matterbridge. 
It actually supports all shelly BLU included.

Your setup is clearly super complicated.

I imagine the issue is the mdns.

Too many devices on the same host.

I would try to change matter port in Matterbridge 
and see: add -port 5550 to the command line

**Comment by Luligu** [comment]
Or since matterbridge also have matterbridge-zigbee2mqtt plugin… use Matterbridge for everything.

**Comment by Apollon77** [comment]
I think the problem is very simple. Matter by default uses port 5540 ... so if both use Docker host network and both brdges listen on port 5540 then usually only one gets the traffic and not both ... even when both it would end in chaos.

You need to move one to another port.

**Comment by Luligu** [comment]
Hi,
@tammeryousef1006 reported to me your post on the other forum.

I will add the port setup in the matterbridge frontend to make things easier when changing the docker command is not possible.

Same issue when you use Matterbridge HA Add-on.

Sometimes docker configurations are not flexible.

**Comment by Luligu** [comment]
Discord group: https://discord.gg/QX58CDe6hd

**Comment by Luligu** [comment]
## [1.6.6] - 2024-12-06

### Added

- [frontend]: Add the possibility to install a specific version or the dev of any plugin (i.e. you can install matterbridge-hass@dev or matterbridge-hass@0.0.3).
  It is also possible to use the install plugin to install a specific version of matterbridge (i.e. you can install matterbridge@dev or matterbridge@1.6.5)
- [frontend]: Add the possibility to set the matter discriminator for commissioning (you can always override passing **-discriminator [DISCRIMINATOR]** on the command line).
- [frontend]: Add the possibility to set the matter passcode for commissioning (you can always override passing **-passcode [PASSCODE]** on the command line).
- [frontend]: Add the possibility to set the matter port for commissioning (you can always override passing **-port [PORT]** on the command line).
- [deviceTypes]: Add device type airConditioner (not supported by the Apple Home).
- [docker]: Add matterbridge-hass to docker dev.
- [platform]: Added validateDeviceWhiteBlackList and validateEntityBlackList to be use consistently by all plugins.

### Changed

- [package]: Update matter.js to 0.11.9-alpha.0-20241206-22f23333.
- [plugin]: Removed check on package types since we are moving to production plugins.
- [package]: Set required node version to 18, 20 and 22.
- [package]: Update dependencies.
- [levelControl]: Set default to OnOff.Feature.Lighting.
- [jest]: Update Jest tests.

### Fixed

- [device]: Fix typos in Device and Endpoint.

@roger-richards is published under dev tag and even if you don't have a terminal you install it from the frontend.

**Comment by roger-richards** [comment]
Thank you for this quick turnaround.  When I try to add luligu/matterbridge to iHost docker, it is still pulling v 1.6.5
I've tried adding the tag dev or test but iHost docker seems to ignore the tag and I have not found a way.  Is 1.6.6 released to the general "latest" repository?  Maybe there is an alternate repository where I can get 1.6.6?

**Comment by tammeryousef1006** [comment]
i msg you already at discord kindly check

**Comment by Luligu** [comment]
Hi,
there are 2 ways to get 1.6.6:
1) you require the docker image with tag dev instead of latest;
2) from the frontend use this trick: install the plugin matterbridge@dev, wait to see package installed correctly, don’t consider the error messages, then restart.

**Comment by tammeryousef1006** [comment]
![ihost](https://github.com/user-attachments/assets/8167757a-1d05-4b84-a1bb-d73aadd32e40)

kindly write the same in yellow circle at your mb ihost and hit install , after that you will find at setting the option to change port , this is fast solution until it get published

**Comment by Luligu** [comment]
I checked again docker hub and dev is published ...

**Comment by tammeryousef1006** [comment]
solved at latest build v1.6.6 plublished

**Comment by roger-richards** [comment]
I have good news to report!  I installed released 1.6.6 in iHost docker, set port to 5541 and Apple Home added the bridge along with a Shelly device without issue! Thanks for your efforts to resolve the problem.

**Comment by tammeryousef1006** [comment]
you welcome and thanks for feedback
